stable/datadog/templates/agent-scc.yaml

```
@@ -0,0 +1,65 @@
+{{- if .Values.agents.podSecurity.securityContextConstraints.create }}
+kind: SecurityContextConstraints
+apiVersion: security.openshift.io/v1
+metadata:
+  name: {{ template "datadog.fullname" . }}
+  labels:
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    app.kubernetes.io/name: "{{ template "datadog.fullname" . }}"
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+users:
+- system:serviceaccount:{{ .Release.Namespace }}:{{ template "datadog.fullname" . }}
+priority: 10
+# Allow host ports for dsd / trace intake
++allowHostPorts: {{ or .Values.datadog.dogstatsd.useHostPort .Values.datadog.apm.enabled }}
+# Allow host PID for dogstatsd origin detection
+allowHostPID: {{ .Values.datadog.dogstatsd.useHostPID }}
+# Allow host network for the CRIO check to reach Prometheus through localhost
+allowHostNetwork: {{ .Values.agents.useHostNetwork }}
+# Allow hostPath for docker / process metrics
+volumes:
+{{ toYaml .Values.agents.podSecurity.volumes | indent 2 }}
+# Use the `spc_t` selinux type to access the
+# docker/cri socket + proc and cgroup stats
+seLinuxContext:
+{{- if .Values.datadog.securityContext }}
+{{ toYaml .Values.datadog.securityContext | indent 2 }}
+{{- else }}
+  type: MustRunAs
+  seLinuxOptions:
+    user: "system_u"
+    role: "system_r"
+    type: "spc_t"
+    level: "s0"
```

If we explicitly set the default value in `values.yaml`, it would allow us to remove that `if`.
```suggestion
{{ toYaml .Values.datadog.securityContext | indent 2 }}
```